### PR TITLE
MB-8065 External IP logging

### PR DESCRIPTION
**Pull request by nyanbinaryneko**
## Description
Creates a logging function in `pkg/cli/logging.go` that will allow us to log our outbound IP for troubleshooting outbound API calls such as gex. This pings the [AWS IP Check](https://checkip.amazonaws.com/), and parses the response. 

I have added the function to the `init` for `gex_sftp` for testing and helping with logging gex going forward.  

## Reviewer Notes

This seemed the simplest, least painful, way to add external IP logging. 
I moved this to the `logging.go` file in the `cli` directory, if there's a better place for this to go, please let me know!

## Setup

Add any steps or code to run in this section to help others prepare to run your code:
1. create a json file with the following:
```json
{
    "body": {
      "sendToSyncada": true,
      "readFromSyncada": true,
      "deleteFromSyncada": false
    }
}
```
1. with `make server_run`:

```sh
$ prime-api-client --insecure support-reviewed-payment-requests --filename readfromsyncada.json
```
1. Check to see if there's a `source_address` entry in the logs

## Code Review Verification Steps

* [x] Tested in staging, where gex works.
* [x] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [] Request review from a member of a different team.
* [x] Have the Jira acceptance criteria been met for this change?

## References

* [MB-8065](https://dp3.atlassian.net/browse/MB-8065?atlOrigin=eyJpIjoiY2JmZWU1NjdjOWE4NGUyN2EzMTk2OTdhYzUzOGIzYzgiLCJwIjoiaiJ9)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -57,6 +58,7 @@ require (
 	github.com/pkg/errors v0.9.1
 	github.com/pkg/sftp v1.13.2
 	github.com/pterm/pterm v0.12.27
+	github.com/rdegges/go-ipify v0.0.0-20150526035502-2d94a6a86c40
```
Comment (ahobson) [inline go.mod:61]: Are we ok with using an external service?

File: pkg/cli/gex_sftp.go [hunk 2]
```diff
@@ -71,11 +74,17 @@ func InitGEXSSH(v *viper.Viper, logger Logger) (*ssh.Client, error) {
 	remote := v.GetString(GEXSFTPIPAddressFlag)
 	port := v.GetString(GEXSFTPPortFlag)
 
-	ip, err := ipify.GetIp()
+	resp, err := http.Get("https://checkip.amazonaws.com")
```
Comment (ahobson) [inline pkg/cli/gex_sftp.go:77]: Could/should this be a configurable endpoint? Any chance it might be different for govcloud?

**Review by gidjin — APPROVED** [review]
Changes look good locally, I see my own external ip as the source ip. Will this work in an ecs scheduled task? The code you've modified is called in the `process-edi` scheduled task.

**Comment by nyanbinaryneko** [comment]
> 
> 
> Changes look good locally, I see my own external ip as the source ip. Will this work in an ecs scheduled task? The code you've modified is called in the `process-edi` scheduled task.

I tested on staging where this works, and you do see it in the logs for the ecs tasks, and it was one of our external ips.

**Pull request merged**